File: Benchmarking/30 Euler/euler47/euler47.py
import sys

cons = int(sys.argv[1])
limit = int(sys.argv[2])

def numFactors(number, primeList):
  nod = 0
  pf = False
  remain = number
  for i in range(0, len(primeList)):
    if primeList[i] * primeList[i] > number:
      nod += 1
      return nod
    pf = False
    while remain % primeList[i] == 0:
      pf = True
      remain = remain / primeList[i]
    if pf:
      nod += 1
    if remain == 1:
      return nod
  return nod

def getPrimes(size):
	array = [True for i in range(size+1)]
	c = 1
	for i in range(2, size):
		if array[i]:
			j = i + i
			while j <= size:
				array[j] = False
				j += i
			c += 1
	primeArray = []
	for i in range(2, size+1):
		if(array[i]):
			primeArray.append(i)
	return primeArray

def calc(limit, consec):
  primeList = getPrimes(limit)
  targetpf = consec
  targetcons = consec
  cons = 1
  result = 1
  while cons < targetcons:
    result += 1
    if numFactors(result, primeList) >= targetpf:
      cons += 1
    else:
      cons = 0
  return result-3

print calc(limit, cons)

File: Benchmarking/30 Euler/euler47/euler47.java
import java.util.*;

public class euler47
{
	public static void main(String[] args)
	{
		int cons = Integer.parseInt(args[0]);
		int limit = Integer.parseInt(args[1]);
		calc(limit, cons);
	}
	
	private static int numFactors(int number, List<Integer> primeList)
	{
		int nod = 0;
		boolean pf = false;
		int remain = number;
		for(int i = 0; i < primeList.size(); i++)
		{
			if(primeList.get(i) * primeList.get(i) > number)
			{
				nod += 1;
				return nod;
			}
			pf = false;
			while(remain % primeList.get(i) == 0)
			{
				pf = true;
				remain = remain / primeList.get(i);
			}
			if(pf)
			{
				nod += 1;
			}
			if(remain == 1)
			{
				return nod;
			}
		}
		return nod;
	}
	
	private static List<Integer> getPrimes(int size)
	{
		List<Boolean> array = new ArrayList<>();
		for(int i = 0; i <= size; i++)
		{
			array.add(true);
		}
		for(int i = 2; i <= size; i++)
		{
			if(array.get(i))
			{
				int j = i + i;
				while(j <= size)
				{
					array.set(j, false);
					j += i;
				}
			}
		}
		List<Integer> primeArray = new ArrayList<>();
		for(int i = 2; i <= size; i++)
		{
			if(array.get(i))
			{
				primeArray.add(i);
			}
		}
		return primeArray;
	}
	
	private static int calc(int limit, int consec)
	{
		List<Integer> primeList = getPrimes(limit);
		int targetpf = consec;
		int targetcons = consec;
		int cons = 1;
		int result = 1;
		while(cons < targetcons)
		{
			result += 1;
			if(numFactors(result, primeList) >= targetpf)
			{
				cons += 1;
			}
			else
			{
				cons = 0;
			}
		}
		return result-3;
	}
}
